test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -1500,6 +1502,83 @@ public function testGetTraitAliases(): void
         ], $classInfo->getTraitAliases());
     }
 
+    public function testGetTraitNamesWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraitNames();
+    }
+
+    public function testGetTraitsWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraits();
+    }
+
+    public function testGetTraitAliasesWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        self::assertSame(
+            [],
+            $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraitAliases(),
+        );
```

Comment:
`getTraitAliases()` does not require autoloadable symbols, therefore runs without exceptions